blocks/library/embed/index.js

```diff
@@ -73,52 +74,127 @@ registerBlock( 'core/embed', {
 		}
 	},
 
-	edit( { attributes, setAttributes, focus, setFocus } ) {
-		const { url, title, caption } = attributes;
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.doServerSideRender = this.doServerSideRender.bind( this );
+			this.state = {
+				html: '',
+				type: '',
+				error: false,
+				fetching: false,
+			};
+			this.noPreview = [
+				'facebook.com',
+			];
+			if ( this.props.attributes.url ) {
+				// if the url is already there, we're loading a saved block, so we need to render
+				this.doServerSideRender();
+			}
+		}
 
-		if ( ! url ) {
-			return (
-				<Placeholder icon="cloud" label={ wp.i18n.__( 'Embed URL' ) } className="blocks-embed">
-					<input type="url" className="placeholder__input" placeholder={ wp.i18n.__( 'Enter URL to embed here...' ) } />
-					<Button isLarge>
-						{ wp.i18n.__( 'Embed' ) }
-					</Button>
-				</Placeholder>
+		doServerSideRender( event ) {
+			if ( event ) {
+				event.preventDefault();
+			}
+			const { url } = this.props.attributes;
+			const api_url = wpApiSettings.root + 'oembed/1.0/proxy?url=' + encodeURIComponent( url ) + '&_wpnonce=' + wpApiSettings.nonce; // eslint-disable-line no-undef
+
+			this.setState( { error: false, fetching: true } );
+			fetch( api_url, {
+				credentials: 'include',
+			} ).then(
+				( response ) => {
+					response.json().then( ( obj ) => {
+						const { html, type } = obj;
+						if ( html ) {
+							this.setState( { html, type } );
+						} else {
+							this.setState( { error: true } );
+						}
+						this.setState( { fetching: false } );
+					} );
+				}
 			);
 		}
 
-		return (
-			<figure className="blocks-embed">
-				<div className="iframe-overlay">
-					<iframe src={ url } title={ title } />
-				</div>
-				{ ( caption && caption.length > 0 ) || !! focus ? (
-					<Editable
-						tagName="figcaption"
-						placeholder={ wp.i18n.__( 'Write caption…' ) }
-						value={ caption }
-						focus={ focus }
-						onFocus={ setFocus }
-						onChange={ ( value ) => setAttributes( { caption: value } ) }
-						inline
-						inlineToolbar
-					/>
-				) : null }
-			</figure>
-		);
+		render() {
+			const { html, type, error, fetching } = this.state;
+			const { url, caption } = this.props.attributes;
+			const { setAttributes, focus, setFocus } = this.props;
+
+			if ( ! html ) {
+				return (
+					<Placeholder icon="cloud" label={ wp.i18n.__( 'Embed URL' ) } className="blocks-embed">
+						<form onSubmit={ this.doServerSideRender }>
+							<input
+								type="url"
+								className="components-placeholder__input"
```

Comment:
If we are to adhere to the proposed guideline, this should be `blocks-` prefix, not `components-`

https://github.com/WordPress/gutenberg/blob/master/docs/coding-guidelines.md#naming